src/TableauConnector.js

```diff
@@ -73,8 +79,21 @@ class TableauConnector {
     if (this.code && tableau.phase !== tableau.phaseEnum.gatherDataPhase) {
       utils.log('SUCCESS: Authenticate (oauth)')
       const code = this.code
-      this.code = null // Code can only be used once
-      return auth.getToken(code)
+      return auth.getToken(code).then(accessToken => {
+        const {dataset_name, query, queryType, forceTableau} = this.params
+        // Restore canonical WDC URL, which Tableau saves with data source
+        let canonicalUrl = `/?dataset_name=${encodeURIComponent(dataset_name)}`
```

Comment:
Defaults to `dataset_name=undefined` when the user had not provided a dataset name e.g. `http://tableau.data.world` instead of `http://tableau.data.world/?dataset_name=kinuthian/tesr`

We should first check whether the variable is defined.

<img width="806" alt="screen shot 2018-07-31 at 21 08 48" src="https://user-images.githubusercontent.com/17295379/43478653-1b623f58-9507-11e8-880a-99fdaad77aed.png">